src/systems/systems/src/hydraulic/mod.rs

```diff
@@ -1668,6 +1676,10 @@ impl Section {
         if let Some(valve) = &self.leak_measurement_valve {
             self.pressure_switch
                 .update(context, valve.downstream_pressure());
+
+            if let Some(priority_valve) = &mut self.priority_valve {
+                priority_valve.update(context, valve.downstream_pressure())
+            }
```

Comment:
Are we sure this should only be called if a leak measurement valve exists? Does this assume that there can only be a priority valve if there's a leak measurement valve?

Also, I think we can move the pressure switch update call outside of the `if let Some(valve) = &self.leak_measurement_valve` by simply updating it with `self.pressure_downstream_leak_valve()`. The if statement is already there. This might be outside of the scope of this PR though.

Reply:
So last time I tried changing stuff around this I broke something for Luke. And in the meantime your point seems totally valid and furthermore current implementation is just wrong.....  Priority valve is not downstream of  leak measurement valves, so flaps for example should still move if  leak measurement valve is closed.